### ISSUE TITLE
test_fontconfig_fonts error on Linux wheel testing

### ISSUE THREAD

**Issue by matthew-brett** [matplotlib/matplotlib#8152]
Testing the built wheel, I get the following error (using pytest):

```
=================================== FAILURES ===================================
__________________________ test_get_fontconfig_fonts ___________________________
[gw0] linux2 -- Python 2.7.6 /venv/bin/python2.7
@pytest.mark.skipif(sys.platform == 'win32', reason='no fontconfig on Windows')
    def test_get_fontconfig_fonts():
>       assert len(get_fontconfig_fonts()) > 1
E       assert 0 > 1
E        +  where 0 = len([])
E        +    where [] = get_fontconfig_fonts()
```

https://travis-ci.org/MacPython/matplotlib-wheels/jobs/205418240#L623

Should a bare Linux container like the testing container contain any fontconfig fonts?  Or is some assumption being made about the configuration of the system?

**Comment by tacaswell** [comment]
Blah, this test went into to verify that we can parse the output from fontconfig properly.

It is reasonable to expect that a testing system have at least one system-level font installed.

**Comment by QuLogic** [comment]
Does `fc-list --format='%{file}\n'` output anything? Does `fc-list` exist on OSX (or does it need homebrew)?

**Comment by QuLogic** [comment]
Oops, this is trusty, not OSX; yea I guess maybe a system font should exist there....

**Comment by matthew-brett** [comment]
`fc-list` is not installed on the trusty container.

**Comment by matthew-brett** [comment]
Reasonable to skip this test when `fontconfig` / `fc-list` not installed?

**Comment by QuLogic** [comment]
Interestingly, fontconfig is not listed in the dependencies, optional or not, even though it's used by `FontManager` regardless of the `USE_FONTCONFIG` setting. Do the rest of the tests run correctly even without fontconfig?

**Comment by matthew-brett** [comment]
They appear to : https://travis-ci.org/MacPython/matplotlib-wheels/jobs/205423488

**Comment by QuLogic** [comment]
Yes, but:
> !!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!!
> ======== 1 failed, 840 passed, 221 skipped, 8 xfailed in 189.47 seconds ========

That's not even close to all the tests.

**Comment by matthew-brett** [comment]
Right - but for the 2.0.0 builds, on the same container, the tests did run : https://travis-ci.org/MacPython/matplotlib-wheels/jobs/195613539 .  So, at least for 2.0.0, I believe fontconfig was not required.

**Comment by tacaswell** [comment]
We hit fontconfig (if it is available) and then also search our internal data folders.  If the call to `fc-list` fails, we just treat that as it returning [], thus `fc-list` is optional, but should be required for testing.

**Comment by anntzer** [comment]
Closed by #8213, afaict.
